com.unity.shadergraph/Editor/GraphUI/ShaderGraphMainToolbar.cs

```diff
@@ -1,11 +1,135 @@
+            Precision,
+            UserDefined
+        }
+        public new static readonly string ussClassName = "ge-sg-toolbar";
+
+        protected ToolbarButton m_SaveButton;
+        protected ToolbarButton m_SaveAsButton;
+        protected ToolbarButton m_ShowInProjectButton;
+        protected ToolbarButton m_CheckOutButton;
+        protected EnumField m_ColorModeDropdown;
+        protected ToolbarButton m_ToggleBlackboardButton;
+        protected ToolbarButton m_ToggleInspectorButton;
+        protected ToolbarButton m_ToggleMainPreviewButton;
+
+        public static readonly string SaveButton = "saveButton";
+        public static readonly string SaveAsButton = "saveAsButton";
+        public static readonly string ShowInProjectButton = "showInProjectButton";
+        public static readonly string CheckOutButton = "checkOutButton";
+        public static readonly string Spacer = "flexibleSpacer";
+        public static readonly string ColorModeDropdown = "colorModeDropdown";
+        public static readonly string ToggleBlackboardButton = "toggleBlackboardButton";
+        public static readonly string ToggleGraphInspectorButton = "toggleGraphInspectorButton";
+        public static readonly string ToggleMainPreviewButton = "toggleMainPreviewButton";
+
+
         public ShaderGraphMainToolbar(BaseGraphTool graphTool, GraphView graphView)
             : base(graphTool, graphView)
+        {
+            // Get rid of all existing visual elements defined by MainToolbar and add them in order we need here
+            Clear();
+
+            AddToClassList(ussClassName);
+
+            this.AddStylesheet("ShaderGraphMainToolbar.uss");
+            this.AddStylesheet(EditorGUIUtility.isProSkin ? "ShaderGraphMainToolbar_dark.uss" : "ShaderGraphMainToolbar_light.uss");
+
+            var tpl = GraphElementHelper.LoadUXML("ShaderGraphMainToolbar.uxml");
+            tpl.CloneTree(this);
+
+            m_SaveButton = this.MandatoryQ<ToolbarButton>(SaveButton);
+            m_SaveButton.ChangeClickEvent(OnSaveButton);
+            Add(m_SaveButton);
+
+            m_SaveAsButton = this.MandatoryQ<ToolbarButton>(SaveAsButton);
+            m_SaveAsButton.ChangeClickEvent(OnSaveAsButton);
+            Add(m_SaveAsButton);
+
+            m_ShowInProjectButton = this.MandatoryQ<ToolbarButton>(ShowInProjectButton);
+            m_ShowInProjectButton.ChangeClickEvent(OnShowInProjectButton);
+            Add(m_ShowInProjectButton);
+
+            //m_CheckOutButton = this.MandatoryQ<ToolbarButton>(CheckOutButton);
+            //m_CheckOutButton.ChangeClickEvent(OnCheckOutButton);
+            //Add(m_CheckOutButton);
+
+            Add(new ToolbarSpacer());
+
+            Add(m_Breadcrumb);
+
+            var flexibleSpacer = this.MandatoryQ<ToolbarSpacer>(Spacer);
+            Add(flexibleSpacer);
+
+            //Add(new Label("Color Mode"));
+
+            //m_ColorModeDropdown = this.MandatoryQ<EnumField>(ColorModeDropdown);
```

Comment:
Are these comments still useful?

Reply:
the intent here is to uncomment these in the future whenever we enable these bits and pieces of the top toolbar, I just wanted to figure out the layout for the top toolbar as a whole now, and then expose the bits that we actually need for our sprint targets over time

Reply:
Okay. Can you please add a comment saying "these should be uncommented when we add the functionality" or "migration - uncomment when functionality available" or something like that?

Reply:
will do!